kyaml/yaml/rnode.go

```diff
@@ -248,6 +377,619 @@ func (rn *RNode) IsNilOrEmpty() bool {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("can't set line comment because the field doesn't exist")
+	}
+	node.YNode().LineComment = comment
+	return nil
+}
+
+// SetHeadComment sets the head comment of the field.
+func (o *KubeObject) SetHeadComment(comment string, fields ...string) error {
+	node := &RNode{}
+	found, err := o.Get(node, fields...)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("can't set head comment because the field doesn't exist")
+	}
+	node.YNode().HeadComment = comment
+	return nil
+}
+
+// RemoveOrDie removes the field located by fields if found. It will panic if it
+// encounters any error.
+func (o *KubeObject) RemoveOrDie(fields ...string) {
+	if err := o.Remove(fields...); err != nil {
+		panic(err)
+	}
+}
+
+// Remove removes the field located by fields if found. It returns if the field
+// is found and a potential error.
+func (o *KubeObject) Remove(fields ...string) error {
+	if o.isNil() {
+		return fmt.Errorf("the object doesn't exist")
+	}
+	err := func() error {
+		parent := o.AsRNode()
+		var found bool
+		var err error
+		if len(fields) > 1 {
+			parent, found, err = o.nestedValue(fields[:len(fields)-1]...)
+			if err != nil {
+				return err
+			}
+			if !found {
+				return nil
+			}
+		}
+
+		_, err = FieldClearer{
+			Name: fields[len(fields)-1],
+		}.Filter(parent)
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+	if err != nil {
+		return fmt.Errorf("unable to remove fields %v with error: %w", fields, err)
+	}
+	return nil
+}
+
+// KubeObjectsToYAML encodes a list of KubeObject as yaml. The objects are
+// seperated by "---"
+func KubeObjectsToYAML(objs []KubeObject) (string, error) {
+	out, er := func() (string, error) {
+		var buf bytes.Buffer
+		isFirst := true
+		for _, obj := range objs {
+			if !isFirst {
+				if _, err := buf.WriteString("---\n"); err != nil {
+					return "", err
+				}
+			}
+			isFirst = false
+			s, err := obj.ToYAML()
+			if err != nil {
+				return "", err
+			}
+			if _, err = buf.WriteString(s); err != nil {
+				return "", err
+			}
+		}
+		return buf.String(), nil
+	}()
+	if er != nil {
+		return "", fmt.Errorf("unable to convert KubeObjects to yaml: %w", er)
+	}
+	return out, nil
+}
+
+// ToYAML encodes the KubeObject as yaml.
+func (o *KubeObject) ToYAML() (string, error) {
+	if o == nil {
+		return "", fmt.Errorf("the object doesn't exist")
+	}
+	return o.AsRNode().String()
+}
+
+// cleanupCreationTimestamp tries to remove field metadata.creationTimestamp. If
+// it encounters any error, it aborts.
+func (o *KubeObject) cleanupCreationTimestamp() {
+	if o.rNode.YNode().Kind != yaml.MappingNode {
+		return
+	}
+	scalar, found, err := o.nestedScalar("metadata", "creationTimestamp")
+	if err != nil || !found {
+		return
+	}
+	if scalar.YNode().Tag == NodeTagNull {
+		o.Remove("metadata", "creationTimestamp")
+	}
+}
+
+// GetMeta returns the ResourceMeta for a KubeObject
+func (o *KubeObject) GetMeta() (ResourceMeta, error) {
```

Comment:
Do you mean `Meta`? The naming conventions seems not aligned:
RNode `GetKind` `GetMeta`, KubeObj `Kind` *Get*Meta

Reply:
Updated